### PR TITLE
[DOC] Fix arg parser

### LINKED ISSUE #194: Required arguments listed as optional in CLI
This is an unfortunate side effect of having our required arguments as options (i.e., with flags instead of positional args). This only affects the help string and not behavior in any way, so it is of very low importance. That said, [this workaround](https://stackoverflow.com/a/43456577/2589328) seems to work well.

Current help message:
```
usage: tedana [-h] -d FILE [FILE ...] -e TE [TE ...] [--mask FILE]
              [--mix FILE] [--ctab FILE] [--manacc MANACC] [--sourceTEs STE]
              [--combmode {t2s,ste}] [--verbose] [--tedort]
              [--gscontrol {t1c,gsr} [{t1c,gsr} ...]] [--wvpca]
              [--tedpca {mle,kundu,kundu-stabilize}] [--out-dir OUT_DIR]
              [--seed FIXED_SEED]

optional arguments:
  -h, --help            show this help message and exit
  -d FILE [FILE ...]    Multi-echo dataset for analysis. May be a single file
                        with spatially concatenated data or a set of echo-
                        specific files, in the same order as the TEs are
                        listed in the -e argument.
  -e TE [TE ...]        Echo times (in ms). E.g., 15.0 39.0 63.0
  --mask FILE           Binary mask of voxels to include in TE Dependent
                        ANAlysis. Must be in the same space as `data`.
  --mix FILE            File containing mixing matrix. If not provided, ME-PCA
                        & ME-ICA is done.
  --ctab FILE           File containing a component table from which to
                        extract pre-computed classifications.
  --manacc MANACC       Comma separated list of manually accepted components
  --sourceTEs STE       Source TEs for models. E.g., 0 for all, -1 for opt.
                        com., and 1,2 for just TEs 1 and 2. Default=-1.
  --combmode {t2s,ste}  Combination scheme for TEs: t2s (Posse 1999, default),
                        ste (Poser)
  --verbose             Generate intermediate and additional files.
  --tedort              Orthogonalize rejected components w.r.t. accepted
                        components prior to denoising.
  --gscontrol {t1c,gsr} [{t1c,gsr} ...]
                        Perform additional denoising to remove spatially
                        diffuse noise. Default is None. This argument can be
                        single value or a space delimited list
  --wvpca               Perform PCA on wavelet-transformed data
  --tedpca {mle,kundu,kundu-stabilize}
                        Method with which to select components in TEDPCA
  --out-dir OUT_DIR     Output directory.
  --seed FIXED_SEED     Value passed to repr(mdp.numx_rand.seed()) Set to an
                        integer value for reproducible ICA results; otherwise,
                        set to -1 for varying results across calls.
```

Desired help message:
```
usage: tedana [-h] -d FILE [FILE ...] -e TE [TE ...] [--mask FILE]
              [--mix FILE] [--ctab FILE] [--manacc MANACC] [--sourceTEs STE]
              [--combmode {t2s,ste}] [--verbose] [--tedort]
              [--gscontrol {t1c,gsr} [{t1c,gsr} ...]] [--wvpca]
              [--tedpca {mle,kundu,kundu-stabilize}] [--out-dir OUT_DIR]
              [--seed FIXED_SEED]

required arguments:
  -d FILE [FILE ...]    Multi-echo dataset for analysis. May be a single file
                        with spatially concatenated data or a set of echo-
                        specific files, in the same order as the TEs are
                        listed in the -e argument.
  -e TE [TE ...]        Echo times (in ms). E.g., 15.0 39.0 63.0

optional arguments:
  -h, --help            show this help message and exit
  --mask FILE           Binary mask of voxels to include in TE Dependent
                        ANAlysis. Must be in the same space as `data`.
  --mix FILE            File containing mixing matrix. If not provided, ME-PCA
                        & ME-ICA is done.
  --ctab FILE           File containing a component table from which to
                        extract pre-computed classifications.
  --manacc MANACC       Comma separated list of manually accepted components
  --sourceTEs STE       Source TEs for models. E.g., 0 for all, -1 for opt.
                        com., and 1,2 for just TEs 1 and 2. Default=-1.
  --combmode {t2s,ste}  Combination scheme for TEs: t2s (Posse 1999, default),
                        ste (Poser)
  --verbose             Generate intermediate and additional files.
  --tedort              Orthogonalize rejected components w.r.t. accepted
                        components prior to denoising.
  --gscontrol {t1c,gsr} [{t1c,gsr} ...]
                        Perform additional denoising to remove spatially
                        diffuse noise. Default is None. This argument can be
                        single value or a space delimited list
  --wvpca               Perform PCA on wavelet-transformed data
  --tedpca {mle,kundu,kundu-stabilize}
                        Method with which to select components in TEDPCA
  --out-dir OUT_DIR     Output directory.
  --seed FIXED_SEED     Value passed to repr(mdp.numx_rand.seed()) Set to an
                        integer value for reproducible ICA results; otherwise,
                        set to -1 for varying results across calls.
```

### PR DESCRIPTION
**Pull request by dowdlelt**
Now required argument show up when tedana or t2smap help is called, via tedana -h/ t2smap -h

Closes #194 .

Changes proposed in this pull request:

-required arguments show up first, in required group, when viewing tedana help
-Same solution used for t2smap
-took directly from the ~~StackExchange~~ stackoverflow - https://stackoverflow.com/a/43456577

### CODE REVIEW

**Comment by dowdlelt** [comment]
Scratch that - help works great~... but tedana fails. I'll investigate a bit more.

File: tedana/workflows/tedana.py [hunk 1]
```diff
@@ -27,7 +27,9 @@ def _get_parser():
     parser.parse_args() : argparse dict
     """
     parser = argparse.ArgumentParser()
-    parser.add_argument('-d',
+    optional = parser._action_groups.pop()
```
Comment (tsalo) [inline tedana/workflows/tedana.py]: Could you maybe add a comment with the link to the SO answer so people know why we are taking this unique approach?

Comment (dowdlelt) [reply tedana/workflows/tedana.py]: Absolutely. Done

**Comment by tsalo** [comment]
Overall this looks good, but could you do a couple of things:
- Fix the spacing (this is just because `required` and `optional` are longer than `parser`)
- Update `t2smap` as well
- Add the [DOC] label to the PR title

**Comment by dowdlelt** [comment]
It took me a hot minute to figure out what was meant by spacing - lets see if I guess correctly.

**Comment by emdupre** [comment]
Looks great, thanks @dowdlelt !!

**Pull request merged**